packages/core/src/render3/instructions/shared.ts

```diff
@@ -1279,13 +1282,13 @@ function findDirectiveDefMatches(
                 `"${tNode.value}" tags cannot be used as component hosts. ` +
                     `Please use a different tag to activate the ${stringify(def.type)} component.`);
 
-            if (tNode.flags & TNodeFlags.isComponentHost) {
+            if (tNode.componentOffset > -1) {
```

Comment:
nit: should we consider using the `isComponentHost` helper function here to make the code more self-documenting?
```suggestion
            if (isComponentHost(tNode)) {
```
This should be ok in this case as it's an ngDevMode-only assertion, but we should consider perf implications if we decide to use the `isComponentHost` in other places below.